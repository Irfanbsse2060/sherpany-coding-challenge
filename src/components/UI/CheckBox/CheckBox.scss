@import "../../../styles/variables";

.check-box {
  list-style: none;
  padding: 0;
  margin: 0;

  &__item {
    padding: 20px;
    display: grid;
    grid-template-columns: 50px auto;

    input {
      cursor: pointer;
    }

  }


}
