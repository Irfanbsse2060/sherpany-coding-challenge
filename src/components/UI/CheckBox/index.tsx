import React, {ChangeEvent} from 'react'
import './CheckBox.scss'

interface CheckBoxOptions {
    label: string,
    name: string,
    value: string,
    isChecked: boolean
}


interface CheckBoxProps {
    options: CheckBoxOptions[]
    testId?: string
    handleChange: (name: string, isChecked: boolean) => void
}

export default function CheckBox(props: CheckBoxProps) {
    const {testId, options, handleChange} = props

    const handleClick = (e: any) => {
        handleChange(e.target.name, e.target.checked)
    }

    return (
        <ul className='check-box' data-testid={testId || `check-box`}>
            {
                options.map(({value, label, isChecked, name}) => {
                    return (
                        <li className='check-box__item' key={name}>
                            <input onChange={handleClick}
                                   type="checkbox"
                                   checked={isChecked}
                                   name={name}
                                   id={name}
                                   value={value}/>
                            <label htmlFor={name}>{label}</label>
                        </li>
                    )
                })
            }
        </ul>
    )

}
