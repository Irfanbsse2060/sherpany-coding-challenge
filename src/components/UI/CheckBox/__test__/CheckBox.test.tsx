import React from 'react';
import {render} from '@testing-library/react';
import CheckBox from '../index'


describe('Check Box', () => {

    const options = [
        {isChecked: true, label: 'Canada (CH)', name: 'CH', value: 'CH'},
        {isChecked: false, label: 'Spain (ES)', name: 'ES', value: 'ES'},
        {isChecked: false, label: 'France (FR)', name: 'FR', value: 'FR'}
    ]

    test('should be able to render', () => {
        const {container} = render(<CheckBox options={options} handleChange={jest.fn()}/>);
        expect(container).toMatchSnapshot()
    })

    test('should call handle change method', () => {
        const mockHandleChange = jest.fn()
        const {getByText} = render(<CheckBox options={options} handleChange={mockHandleChange}/>);
        getByText('Spain (ES)').click()
        expect(mockHandleChange).toHaveBeenCalledTimes(1)
        expect(mockHandleChange).toHaveBeenCalledWith("ES", true)
    });

})
