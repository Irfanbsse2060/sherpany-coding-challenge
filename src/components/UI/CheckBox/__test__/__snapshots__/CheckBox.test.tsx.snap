// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Box should be able to render 1`] = `
<div>
  <ul
    class="check-box"
    data-testid="check-box"
  >
    <li
      class="check-box__item"
    >
      <input
        checked=""
        id="CH"
        name="CH"
        type="checkbox"
        value="CH"
      />
      <label
        for="CH"
      >
        Canada (CH)
      </label>
    </li>
    <li
      class="check-box__item"
    >
      <input
        id="ES"
        name="ES"
        type="checkbox"
        value="ES"
      />
      <label
        for="ES"
      >
        Spain (ES)
      </label>
    </li>
    <li
      class="check-box__item"
    >
      <input
        id="FR"
        name="FR"
        type="checkbox"
        value="FR"
      />
      <label
        for="FR"
      >
        France (FR)
      </label>
    </li>
  </ul>
</div>
`;
